ecl/hql/hqlgram2.cpp

```diff
@@ -2643,6 +2643,40 @@ void HqlGram::checkDefaultValueVirtualAttr(const attribute &errpos, IHqlExpressi
     }
 }
 
+void HqlGram::checkAlienType(const attribute &errpos, IHqlAlienTypeInfo * alien)
+{
+    IHqlExpression * store = alien->queryStoreFunction();
+    assertex(store->isFunction());
+    ITypeInfo * type = store->queryType()->queryChildType();
+    if (!isUnknownSize(type))
+        return;
+
+    IHqlExpression * size = alien->queryLengthFunction();
+    if (size->isConstant())
+        return;
+
+    if (size->getOperator() == no_funcdef)
+        size = size->queryChild(0);
+    size = queryStripCasts(size);
+    if (size->getOperator() == no_select)
+    {
+        if (size->queryChild(0)->getOperator() == no_selfref)
+        {
+            IHqlExpression & topRecord = activeRecords.tos();
+            if (topRecord.numChildren())
+            {
+                IHqlExpression * lastField = topRecord.queryChild(topRecord.numChildren()-1);
+                if (lastField == size->queryChild(1))
+                {
+                    if (lastField->queryType()->getSize() == 4)
```

Comment:
Feels like it should be checking that it's a little-endian unsigned integer as well as that it's 4-bytes? Possibly could also accept signed...

Reply:
It depends if you are looking for binary compatibility, or equivalence.  The row translation will make life interesting even if binary compatible.